resources/js/processes-catalogue/components/charts/ListChart.vue

```diff
@@ -0,0 +1,658 @@
+              resolve(this.pages[uid]);
+            });
+          });
+        } else {
+          return this.preloads[uid];
+        }
+      }
+    },
+    preload(uid) {
+      const currentPage = this.pageFromUid(uid);
+      const nextPage = currentPage + 1;
+      const nextUid = this.uid(nextPage, this.filterFromUid(uid));
+
+      if (nextPage <= this.totalPages) {
+        if (!this.pages[nextUid] && !this.preloads[nextUid]) {
+          this.preloads[nextUid] = this.getPage(nextUid);
+        }
+      }
+    },
+    onRowClicked(record, index, event) {
+      if (record.ProcessMaker__url) {
+        if (!this.config.display.pivot && this.config.display.linkRow) {
+          let target = '_self';
+          if (event.metaKey || event.ctrlKey) {
+            target = '_blank';
+          }
+          window.open(record.ProcessMaker__url, target);
+        }
+      }
+    },
+    onRowMiddleClicked(record, index, event) {
+      if (record.ProcessMaker__url) {
+        if (!this.config.display.pivot && this.config.display.linkRow) {
+          window.open(record.ProcessMaker__url, '_blank');
+        }
+      }
+    },
+    pageUrl(page) {
+      let url = `${this.url}?` +
+                `data_page=${page}&` +
+                `data_per_page=${this.perPage}&` +
+                `data_order_by=${encodeURIComponent(this.orderBy)}&` +
+                `data_order_direction=${this.orderDirection}&` +
+                `data_filter=${this.dataFilter}&` +
+                `data_pmql=${this.dataPmql}`;
+
+      if (this.additionalPmql && this.additionalPmql.length) {
+        url += `&additional_pmql=${this.additionalPmql}`;
+      }
+
+      return url;
+    },
+    clear() {
+      this.pages = {};
+      this.preloads = {};
+      this.currentPage = 1;
+    },
+    render() {
+      if (! this.preview) {
+        this.renderChart(this.data, this.options);
+      } else {
+        this.renderPreview();
+      }
+      this.$emit('render');
+    },
+    describe() {
+      return this.$t('Data Table');
+    },
+    renderPreview() {
+      this.rendered = true;
+    },
+    renderChart(data, options) {
+      this.rendered = false;
+      this.$nextTick(() => {
+        this.chartData = data;
+        this.chartOptions = options;
+        this.originalData = JSON.stringify(data);
+        this.rendered = true;
+      });
+    },
+    onInputResize(size) {
+      this.pmqlInputHeight = size;
+    },
+    onNLQConversion(pmql) {
+      this.filter = pmql;
+    },
+    runSearch() {
+      this.filter = this.query;
+    },
+  },
+  watch: {
+    data: {
+      handler: function(value) {
+        if (JSON.stringify(value) !== this.originalData) {
+          this.clear();
+          this.render();
+        }
+      },
+      deep: true
+    },
+    filter(filter) {
+      this.currentPage = 1;
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+  @import "../../../../../vendor/processmaker/package-savedsearch/resources/sass/_variables.scss";
```

Comment:
this will bring some errors

Reply:
This line(similar path) originally bring errors during compile because all this file was copied from SavedSearch package. Currently code works fine.